### PR TITLE
schelp `Object.dump` update: more explanation with case examples

### PR DESCRIPTION
**Pull request by prko**
## Purpose and Motivation
The explanation of .dump is too minimal and abstract. I have summarised what I learned from scsynth.org and the old mailing list.
In particular, the following thread was helpful:

- https://scsynth.org/t/understanding-the-result-of-dump-low-level-data-for-float-hexadecimal-for-string-gc-fmt-flg-and-set/8553/3

- https://listarc.cal.bham.ac.uk/lists/sc-users-2016/msg50531.html

Thanks to @telephon, @dkmayer, @adcxyz, @jamshark70, and @mossheim!

I have added explanations that are as detailed as possible and as simple as possible for those new to SuperCollider with no programming experience.

I hope this can be approved and merged if the content is correct. 
In the hope that someone will let me correct the wrong information rather than be ignored, so that this PR can help someone like me without having to search the internet.

PDF files of the corresponding parts in the object help file:
- [Object _ SuperCollider 3.14.0-dev Help - with custom css.pdf](https://github.com/supercollider/supercollider/files/15208545/Object._.SuperCollider.3.14.0-dev.Help.-.with.custom.css.pdf)
- [Object _ SuperCollider 3.14.0-dev Help - with default css.pdf](https://github.com/supercollider/supercollider/files/15208546/Object._.SuperCollider.3.14.0-dev.Help.-.with.default.css.pdf)

## Types of changes

- Documentation

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: HelpSource/Classes/Object.schelp [hunk 1]
```diff
@@ -692,8 +692,129 @@ code::
 method::dump
 
 Print a detailed low level representation of the receiver to the post window.
+Except for the link::Classes/List:: class, this method is not overridden in other classes. Any of the MetaClasses, Classes and Instances can be used by this method.
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:695]: suggestions follow.

Any object understands this method (this means metaclasses, classes and their instances). Note that in link::Classes/List:: class, this method dumps the list's internal array.

File: HelpSource/Classes/Object.schelp [hunk 2]
```diff
@@ -692,8 +692,129 @@ code::
...
+::
+teletype::
+   Symbol '1'
+-> 1
+::
+::
+
+## Some instance objects return more detailed information, such as
+list::
+## addrress in virtual memory (the hexadecimal number prefixed with 0x),
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:767]: address

File: HelpSource/Classes/Object.schelp [hunk 3]
```diff
@@ -692,8 +692,129 @@ code::
...
+## emphasis::garage collector color:: (gc),
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:768]: not garage! :)

File: HelpSource/Classes/Object.schelp [hunk 4]
```diff
@@ -692,8 +692,129 @@ code::
...
+## emphasis::data format type:: (fmt),
+## emphasis::flags for immutablity, finalization and garbage collector debug sanity check:: (flg),
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:770]: sanity information from the garbage collector

File: HelpSource/Classes/Object.schelp [hunk 5]
```diff
@@ -692,8 +692,129 @@ code::
...
+Object.dump      // the class called Object
+::
+code::
+Object.new.dump  // an istance of the class Object
+::
+
+discussion::
+
+list::
+## The detailed low level format and information varies depending on the receiver.
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:710]: The detailed low level information varies depending on the receiver.

File: HelpSource/Classes/Object.schelp [hunk 6]
```diff
@@ -692,8 +692,129 @@ code::
...
+## Some instance objects, especially unique objects, return the class name and value (also low data if necessary) of the dumped object:
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:711]: I don't understand: "(also low data if necessary)". Maybe (also more data if necessary) ?

File: HelpSource/Classes/Object.schelp [hunk 7]
```diff
@@ -692,8 +692,129 @@ code::
...
+||
+code::
+1.0.dump
+::
+64-bit version of SuperCollider returns:
+teletype::
+   Float 1.000000   00000000 3FF00000
+-> 1.0
+::
+The last two groups of an 8-digit integer are the raw hexadecimal representation of the 64-bit double value according to link::https://ieeexplore.ieee.org/document/8766229##IEEE 754 Floating Point:: (https://ieeexplore.ieee.org/document/8766229). Each part is represented as follows:
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:724]: This information could go into `Float` help. Here it is perhaps too special and takes much space for that.

**Review by telephon — CHANGES_REQUESTED** [review]

File: HelpSource/Classes/Object.schelp (hunk 1 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done!

File: HelpSource/Classes/Object.schelp (hunk 3 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done! (oh. sorry!)

File: HelpSource/Classes/Object.schelp (hunk 5 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done!

File: HelpSource/Classes/Object.schelp (hunk 6 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Yes, your correction is better!
Thanks, done!

File: HelpSource/Classes/Object.schelp (hunk 7 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done!

I changed `Float` help document in this PR:
https://github.com/supercollider/supercollider/pull/6292/commits/cd5008216c2831587e2c50b645c6c34093ebaeef

File: HelpSource/Classes/Object.schelp (hunk 2 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done! (oh. sorry!)

File: HelpSource/Classes/Object.schelp (hunk 4 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thank you, done! But could you check it again, if I understand you correctly?

File: HelpSource/Classes/Object.schelp [hunk 8]
```diff
@@ -764,10 +755,10 @@ teletype::
 
 ## Some instance objects return more detailed information, such as
```
Comment (telephon) [inline HelpSource/Classes/Object.schelp:756]: just saw this one: "Some instance objects" -> "Some instances"

**Review by telephon — CHANGES_REQUESTED** [review]
Thank you! one more...

File: HelpSource/Classes/Object.schelp (hunk 8 above)
Comment (prko) [reply HelpSource/Classes/Object.schelp]: Thanks, done!

**Review by telephon — APPROVED** [review]

**Pull request merged**